obspy/core/util/decorator.py

```diff
@@ -279,10 +280,17 @@ def new_func(*args, **kwargs):
             # check kwargs
             if arg_kwarg_name in kwargs:
                 if isinstance(kwargs[arg_kwarg_name], (str, native_str)):
-                    if kwargs[arg_kwarg_name].startswith(prefix):
+                    arg_ = kwargs[arg_kwarg_name]
+                    if isinstance(arg_, (bytes, native_bytes)):
+                        # try to find encoding specified in document
+                        match = re.match(r'[^>]* encoding=(["\'])(.+?)\1',
+                                         arg_)
```

Comment:
I'm confused. What happens if the file is not an XML file and thus has no encoding specified?